lib/flex.js

```diff
@@ -0,0 +1,69 @@
+const Configuration = require('./configuration/configuration')
+const extractValidationStats = require('./stats/extractValidationStats')
+
+// Main logic Processor of mergeable
+const executeMergeable = async (context, registry) => {
+  if (registry === undefined) {
+    registry = { validators: new Map(), actions: new Map() }
+  }
+
+  // first fetch the configuration
+  let config = await Configuration.instanceWithContext(context)
+
+  // check config file and get settings only for the current event. the rest is useless for executor
+  let eventName = `${context.event}.${context.payload.action}`
+  await config.filterByEvent(eventName)
```

Comment:
Any reason for filtering events? I propose we don't filter. Here's why:
- If there are no validators that supports an event, the rule will just be ignored. 
- Adding new validators or updating a validator to support new events will require update to the Configuration class filtering logic as well. Coupling all validators to Configuration.